tests/Aql/NoResultsExecutorTest.cpp

```diff
@@ -0,0 +1,148 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2018 ArangoDB GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Jan Christoph Uhde
+////////////////////////////////////////////////////////////////////////////////
+
+#include "BlockFetcherHelper.h"
+#include "catch.hpp"
+
+#include "Aql/AqlItemBlock.h"
+#include "Aql/AqlItemBlockShell.h"
+#include "Aql/ExecutionBlockImpl.h"
+#include "Aql/ExecutorInfos.h"
+#include "Aql/InputAqlItemRow.h"
+#include "Aql/NoResultsExecutor.h"
+#include "Aql/ResourceUsage.h"
+#include "Aql/SingleRowFetcher.h"
+
+#include <velocypack/Builder.h>
+#include <velocypack/velocypack-aliases.h>
+
+using namespace arangodb;
+using namespace arangodb::aql;
+
+namespace arangodb {
+namespace tests {
+namespace aql {
+
+SCENARIO("NoResultsExecutor", "[AQL][EXECUTOR][NORESULTS]") {
```

Comment:
I think there is an important kind of check missing here; namely, that the `NoResultsExecutor` does not consume any input rows. I suggest you add a method to `SingleRowFetcherHelper` which returns `_nrCalled`, and test that it returns `0` after all `produceRow` calls (after the last one in each test would suffice, but to better find the source etc.).